.github/workflows/unit_tests.yml

```diff
@@ -26,24 +26,31 @@ jobs:
         working-directory: .
 
     steps:
-      - uses: actions/checkout@v3
-      - name: Set up Python ${{ matrix.python-version }}
-        uses: actions/setup-python@v4
+      - uses: actions/checkout@v4
+
+      # use github python action instead of uv to take advantage of caching
+      - name: Set up Python ${{ matrix.python }}
+        uses: actions/setup-python@v5
         with:
           python-version: ${{ matrix.python }}
-          cache: 'pip'
-          cache-dependency-path: '**/pyproject.toml'
-      - name: Install package with dependencies
-        run: pip install -e ".[test]"
+
+      - name: Install uv
```

Comment:
We could optionally pin a specific version of `uv`. Sticking with `setup-python` instead of `uv python install` makes sense re: caching.